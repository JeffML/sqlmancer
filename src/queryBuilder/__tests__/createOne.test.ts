import { withDialects } from './__utilties__'
import { ActorCreateOneBuilder } from './__fixtures__/models'

describe('CreateOneBuilder', () => {
  withDialects((options, rollback) => {
    describe('basic queries', () => {
      test('no additional options', async () => {
        const builder = new ActorCreateOneBuilder(options, { firstName: 'SUSAN', lastName: 'ANTHONY' })
        const { sql, bindings } = builder.toQueryBuilder().toSQL()
        await rollback(builder, result => expect(result).toBeNumber())
        expect(sql).toMatchSnapshot()
        expect(bindings).toMatchSnapshot()
      })

      test('non-existent field', async () => {
        const builder = new ActorCreateOneBuilder(options, {
          firstName: 'SUSAN',
          lastName: 'ANTHONY',
          foo: 'bar',
        } as any)
        const { sql, bindings } = builder.toQueryBuilder().toSQL()
        await rollback(builder, result => expect(result).toBeNumber())
        expect(sql).toMatchSnapshot()
        expect(bindings).toMatchSnapshot()
      })
    })
  })
})
