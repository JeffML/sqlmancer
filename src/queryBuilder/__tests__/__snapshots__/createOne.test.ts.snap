// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateOneBuilder pg basic queries no additional options 1`] = `"insert into \\"actor\\" (\\"first_name\\", \\"last_name\\") values (?, ?) returning \\"actor_id\\""`;

exports[`CreateOneBuilder pg basic queries no additional options 2`] = `
Array [
  "SUSAN",
  "ANTHONY",
]
`;

exports[`CreateOneBuilder pg basic queries non-existent field 1`] = `"insert into \\"actor\\" (\\"first_name\\", \\"last_name\\") values (?, ?) returning \\"actor_id\\""`;

exports[`CreateOneBuilder pg basic queries non-existent field 2`] = `
Array [
  "SUSAN",
  "ANTHONY",
]
`;
